### PR TITLE
Add phlippe_resnet

### PR DESCRIPTION
**Pull request by FindHao**
add [phlippe's resnet model](https://github.com/phlippe/uvadlc_notebooks_benchmarking/blob/main/PyTorch/Tutorial5_Inception_ResNet_DenseNet.py).

### CODE REVIEW

File: torchbenchmark/models/phlippe_resnet/__init__.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        super().__init__(test=test, device=device, jit=jit,
+                         batch_size=batch_size, extra_args=extra_args)
+        self.model = ResNetModel()
+        self.model.to(device)
+        self.example_inputs = (
+            torch.randn((self.batch_size, 3, 32, 32), device=self.device),
+        )
+        self.example_target = torch.randint(0, 10, (self.batch_size,), device=self.device)
+        dataset = data.TensorDataset(self.example_inputs[0], self.example_target)
+        dummy_loader = data.DataLoader(dataset, batch_size=self.batch_size)
```
Comment (xuzhao9) [inline torchbenchmark/models/phlippe_resnet/__init__.py:140]: What does dummy_loader do? Can we remove it?

File: torchbenchmark/models/phlippe_resnet/__init__.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        self.optimizer = optim.SGD(self.model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4)
+        self.criterion = nn.CrossEntropyLoss()
+
+    def get_module(self):
+        return self.model, self.example_inputs
+
+    def train(self):
+        model = self.model
+        (images, ) = self.example_inputs
+        model.train()
```
Comment (xuzhao9) [inline torchbenchmark/models/phlippe_resnet/__init__.py:150]: Let's move this to init

File: torchbenchmark/models/phlippe_resnet/__init__.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        output = model(images)
+        loss = self.criterion(output, targets)
+        loss.backward()
+        self.optimizer.step()
+        self.optimizer.zero_grad()
+    
+    def eval(self):
+        model = self.model
+        (images, ) = self.example_inputs
+        model.eval()
```
Comment (xuzhao9) [inline torchbenchmark/models/phlippe_resnet/__init__.py:161]: Let's move this to init

**Review by xuzhao9 — APPROVED** [review]
LGTM with small nits

File: torchbenchmark/models/phlippe_resnet/__init__.py (hunk 2 above)
Comment (FindHao) [reply torchbenchmark/models/phlippe_resnet/__init__.py]: fixed by 70289f205a021189a88f830b7de49d48a44b306e

File: torchbenchmark/models/phlippe_resnet/__init__.py (hunk 3 above)
Comment (FindHao) [reply torchbenchmark/models/phlippe_resnet/__init__.py]: fixed by 70289f205a021189a88f830b7de49d48a44b306e

**Pull request closed without merging**